lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -28,26 +28,34 @@ using llvm::formatv;
 static LogicalResult printConstantOp(CppEmitter &emitter,
                                      ConstantOp constantOp) {
   auto &os = emitter.ostream();
-  if (failed(emitter.emitVariableDeclaration(
-          constantOp.getOperation()->getResult(0),
-          /*trailingSemicolon=*/false)))
-    return failure();
+  auto result = constantOp.getOperation()->getResult(0);
+  auto value = constantOp.value();
 
-  // Add braces for number literals only to avoid double brace intialization for
-  // tensors.
-  auto value = constantOp.getValue();
-  bool emitBraces = value.isa<FloatAttr>() || value.isa<IntegerAttr>();
+  bool isScalar = !result.getType().isa<TensorType>();
 
-  // Replace curly braces with `=` if restricted to C
-  if (emitter.restrictedToC() && emitBraces) {
-    emitBraces = false;
-    os << " = ";
+  // Add braces only if
+  //  - cpp code is emitted,
+  //  - variables are not forward declared
+  //  - and the emitted type is a scalar.
+  bool emitBraces = !emitter.restrictedToC() &&
+                    !emitter.forwardDeclaredVariables() && isScalar;
+
+  // Emit an assignment if variables are forward declared.
+  if (emitter.forwardDeclaredVariables()) {
+    if (failed(emitter.emitVariableAssignment(result)))
+      return failure();
+  } else {
+    if (failed(emitter.emitVariableDeclaration(result,
+                                               /*trailingSemicolon=*/false)))
+      return failure();
+    if (emitter.restrictedToC())
+      os << " = ";
```

Comment:
Why not
```suggestion
    if (emitter.restrictedToC()) {
      os << " = ";
    } else {
      os << "{";
    }
```
Makes somehow more sense to me to put this here and to remove below. At least at this late hour ;)